Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.SwitchExpression.cs

```diff
@@ -49,18 +50,22 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public SwitchExpression Update (Expression switchValue, IEnumerable<SwitchCase> cases, Expression defaultBody)
     {
+      Contract.Requires(switchValue != null);
```

Comment:
Totally agree with your concern. One one hand, I'm sure that static checker would not be able to prove many of them. On the other hand, using Code Contracts Editor Extensions will allow an end user to see them and validate them manually.

I would say that we should use `NotNullOrEmpty` but ommit `Contract.ForAll`...

About `AssumeRequires`: maybe this is a good idea for new feature...